File: tests/src/test/python/test_program.py
# This is intended solely for testing spark-submit as described in the CONTRIBUTING.md file.

from pyspark.sql import SparkSession

SparkSession.builder.getOrCreate().read.format("marklogic") \
  .option("spark.marklogic.client.uri", "spark-test-user:spark@localhost:8016") \
  .option("spark.marklogic.read.opticQuery", "op.fromView('Medical', 'Authors', '')") \
  .load() \
  .show()

File: marklogic-spark-connector/src/test/java/com/marklogic/spark/TestProgram.java
/*
 * Copyright (c) 2023-2025 Progress Software Corporation and/or its subsidiaries or affiliates. All Rights Reserved.
 */
package com.marklogic.spark;

import org.apache.spark.sql.SparkSession;

/**
 * This is only intended for manual testing with spark-submit. As noted in the CONTRIBUTING.md file, you first need to
 * move this to src/main/java and then rebuild the connector jar.
 */
public class TestProgram {

    public static void main(String[] args) {
        SparkSession.builder().getOrCreate().read()
            .format("marklogic")
            .option(Options.CLIENT_URI, "spark-test-user:spark@localhost:8016")
            .option(Options.READ_OPTIC_QUERY, "op.fromView('Medical','Authors')")
            .load()
            .collectAsList()
            .forEach(row -> System.out.println("ROW: " + row.prettyJson()));
    }
}
